File: 543_EllysTSP.py
string = raw_input("Enter the pattern in caps, only letters C and V:")
iterator,c,v = 0,0,0
lst1,lst2,lst3 = [],[],[]
while iterator < len(string):
    if string[iterator] == 'C':
        c+=1
    else:
        v+=1
    iterator+=1
print 'C:',c,'V:',v
iterator = 0
while iterator < len(string):
    if string[iterator] == 'V':
        lst1.append(string[iterator])
    else:
        lst2.append(string[iterator])
    iterator+=1
iterator = 0
if v == c:
    print v+c
if v > c:
    while iterator < len(lst2):
        lst3.append(lst1[iterator])
        lst3.append(lst2[iterator])
        iterator+=1
    lst3.append('V')
iterator = 0
if c > v:
    while iterator < len(lst1):
        lst3.append(lst2[iterator])
        lst3.append(lst1[iterator])
        iterator+=1
    lst3.append('C')
print 'Possible Path:',lst3,'Length:',len(lst3)

File: 543_EllysTSP.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;
class EllysTSP
{
    public:
        int c,v;
        vector <char> lst1;
        vector <char> lst2;
        vector <char> lst3;
        int getMax(string places)
        {
            c=v=0;
            for(int i=0;i<places.size();i++)
            {
                if (places[i] == 'C')
                {
                    c+=1;
                }
                else
                {
                    v+=1;
                }
            }
            for(int i=0;i<places.size();i++)
            {
                if (places[i] == 'V')
                {
                    lst1.push_back(places[i]);
                }
                else
                {
                    lst2.push_back(places[i]);
                }
            }
            if (v>c)
            {
                for(int i=0;i<lst2.size();i++)
                {
                    lst3.push_back(lst1[i]);
                    lst3.push_back(lst2[i]);
                }
                lst3.push_back('V');
            }
            if (c>v)
            {
                for(int i=0;i<lst1.size();i++)
                {
                    lst3.push_back(lst2[i]);
                    lst3.push_back(lst1[i]);
                }
                lst3.push_back('C');
            }
            return lst3.size();
        }
    };
